docs/v1/P4Runtime-Spec.mdk

```diff
@@ -3795,34 +3803,56 @@ the `GetForwardingPipelineConfig RPC`. The request is defined as:
 
 ~ Begin Proto
 message GetForwardingPipelineConfigRequest {
+  enum ResponseType {
+    ALL = 0;
```

Comment:
ALL will be the default value, so a server will see this when the client (say an older version prior to the response type addition) does not specify the type.  In this case, the client probably hasn't provided a cookie.  What should the server do when it has no cookie to return, default the cookie to zero?

Similarly, if a newer client provides a cookie to an older version of the server, the server won't store the cookie, and won't have a cookie to provide in the response.  The server will also not be aware of the response type presence, and thus it may respond with more information than the client expects.  Client and server need to handle these cases for upgrade compatibility.

There is another situation where client and server are both aware of the response type, but SetForwardingPipelineConfig was done before the update occurred, so once again, the server will have no cookie to provide if requested.  All of these situations are leading me to believe that the cookie ought to be wrapped in its own sub-message so that it is obvious whether the server has provided a known value in proto3.

Reply:
Thanks for this comment. I agree on wrapping the cookie in a sub-message. This also helps to clarify the case where a pipeline-config was installed with a mechanism other than `SetForwardingPipelineConfig`, in which case the returned cookie will be unset. In general, I made the use of a cookie field optional.